analysis/descriptives/venn.R

```diff
@@ -250,17 +218,8 @@ df[,setdiff(colnames(df),c("outcome"))] <- lapply(df[,setdiff(colnames(df),c("ou
 
 # Rename columns (output redaction) --------------------------------------------
 
-names(df)[names(df) == "only_snomed"] <- "only_snomed_midpoint6"
-names(df)[names(df) == "only_hes"] <- "only_hes_midpoint6"
-names(df)[names(df) == "only_death"] <- "only_death_midpoint6"
-names(df)[names(df) == "snomed_hes"] <- "snomed_hes_midpoint6"
-names(df)[names(df) == "snomed_death"] <- "snomed_death_midpoint6"
-names(df)[names(df) == "hes_death"] <- "hes_death_midpoint6"
-names(df)[names(df) == "snomed_hes_death"] <- "snomed_hes_death_midpoint6"
-names(df)[names(df) == "total_snomed"] <- "total_snomed_midpoint6"
-names(df)[names(df) == "total_hes"] <- "total_hes_midpoint6"
-names(df)[names(df) == "total_death"] <- "total_death_midpoint6"
-names(df)[names(df) == "total"] <- "total_midpoint6_derived"
+colnames(df)[2:12] <- paste(colnames(df)[2:12], "midpoint6", sep = "_")
+names(df)[names(df) == "total_midpoint6"] <- "total_midpoint6_derived"
```

Comment:
For the total column, the new count would be:
`df$total_midpoint6_derived <- df$total_snomed_midpoint6 + df$total_hes_midpoint6 + df$total_death_midpoint6
`
For `total_snomed`, `total_hes`, and `total_death`, if the counts include the sum of their pairs, it would be something like this (exemple using snomed):
`df$total_snomed <- df$only_snomed_derived + df$snomed_hes_midpoint6 + df$snomed_death_midpoint6`
But this approach will count snomed 3 times (same would apply for the other total columns). Otherwise, I am not sure how do you want to calculate the total columns.

Reply:
For the data source totals, you will not be counting SNOMED three times because individuals can only contribute to one of these categories. As an example, the total for SNOMED is calculated as the sum of only_snomed, snomed_death, snomed_hes and snomed_hes_death.

For the overall total, if you summed total_snomed, total_hes, total_death then you will be counting people multiple times because several of the categories (e.g., snomed_hes_death) contribute to more than one of these totals. You therefore need to calculate the overall total by summing only_death, only_hes, only_snomed, snomed_hes, snomed_death, hes_death, and snomed_hes_death.

Reply:
Following your examples the total columns have been added.

Reply:
Thank you. These columns are correct but the total_snomed, total_hes, total_death need to named 'midpoint6_derived' rather than just 'midpoint6'.